### PR TITLE
[docs] CLI Dataset

### PR DESCRIPTION
**Pull request by sarah-iot**

### CODE REVIEW

**Review by Vyrastas — COMMENTED** [review]
Header files all LGTM, others should be approved by Eng.

File: src/cli/cli_dataset.cpp [hunk 1]
```diff
@@ -118,6 +118,21 @@ otError Dataset::Print(otOperationalDataset &aDataset)
...
+ * @code
+ * dataset init new
+ * Done
+ * @endcode
+ * @cparam dataset init {@ca{active}|@ca{new}|@ca{pending}|@ca{tlvs}} [@ca{hex-encoded-tlvs}]
+ * Use `new` to initialize a new dataset, then enter the command `dataset commit active`.
+ * Use `tlvs` for hex-encoded TLVs.
+ * @par
+ * OT CLI checks for `active`, `pending`, or `tlvs` and returns the corresponding values. Otherwise,
+ * OT CLI creates a new, random network and returns a new dataset. Can be used with OTBR.
```
Comment (jwhui) [inline src/cli/cli_dataset.cpp:132]: Most of these commands can be used on OTBR. What are we trying to convey with "Can be used with OTBR."?

File: src/cli/cli_dataset.cpp [hunk 2]
```diff
@@ -151,6 +166,34 @@ template <> otError Dataset::Process<Cmd("init")>(Arg aArgs[])
...
+ * Network Key: dfd34f0f05cad978ec4e32b0413038ff
+ * Network Name: OpenThread-8f28
+ * PAN ID: 0x8f28
+ * PSKc: c23a76e98f1a6483639b1ac1271e2e27
+ * Security Policy: 0, onrcb
+ * Done
+ * @endcode
+ * @code
+ * dataset active -x
+ * 0e080000000000010000000300001035060004001fffe002084eb74ab03c56e6d00708fdc7fe165c83a67805108e2104f183e698da87e96efc1e45aa51030f4f70656e5468726561642d383631310102861104108d6273023d82c841eff0e68db86f35740c030000ff
```
Comment (jwhui) [inline src/cli/cli_dataset.cpp:187]: This is causing `pretty` check to fail. Is there any way we can include line breaks here that are removed as part of the conversion?

File: src/cli/cli_dataset.cpp (hunk 1 above)
Comment (sarah-iot) [reply src/cli/cli_dataset.cpp]: I removed this...

File: src/cli/cli_dataset.cpp (hunk 2 above)
Comment (sarah-iot) [reply src/cli/cli_dataset.cpp]: I added the common `...` indicator to indicate a trimmed string - will that work?

**Review by Vyrastas — APPROVED** [review]
Header files all LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**